src/lib/testsuite/PostmanResponseChecks.ts

```diff
@@ -0,0 +1,103 @@
+import { OpenApiParser, PostmanParser } from 'application'
+import { Item, Event } from 'postman-collection'
+
+export const generateRequestChecks = (
+  pmObj: PostmanParser,
+  oaObj: OpenApiParser
+): PostmanParser => {
+  // const requests = obj?.item
+
+  // if (!requests) {
+  //   return obj
+  // }
+  // requests.map(obj => injectSkip(obj))
+  // console.log('generateRequestChecks', obj.requests)
+
+  // DEBUG
+  pmObj.requests.length = 1
+  pmObj.mappedOperations.length = 1
+
+  console.log(pmObj.mappedOperations)
+  // console.log('oaObj', oaObj)
+
+  // Iterate over all requests
+  pmObj.mappedOperations.map(pmOperation => {
+    // console.log('pmOperation', pmOperation)
+    // console.log('events', pmOperation.item.events)
+
+    // Get OpenApi responses
+    const oaOperation = oaObj.getOperationByPath(pmOperation.pathRef)
+
+    // Generate response checks
+    generateResponseTests(pmOperation, oaOperation)
+  })
+
+  return pmObj
+}
+
+/**
+ * Generate all portman tests based on the Postman request & the OpenApi specification
+ * @param pmOperation
+ * @param oaOperation
+ */
+const generateResponseTests = (pmOperation: any, oaOperation: any): any => {
+  // console.log('pmRequest', pmRequest)
+  // console.log('oaOperation', oaOperation)
+
+  for (const [code] of Object.entries(oaOperation.schema.responses)) {
+    // // Only support 2xx response checks - Happy path
+    if (!inRange(parseInt(code), 200, 299)) {
+      return // skip this response
+    }
+
+    // Check - Success 2xx response checks
+    generateResponseSuccessCheck(pmOperation, oaOperation)
+  }
+}
+
+/**
+ * Generate response success check
+ * @param pmOperation
+ * @param oaOperation
+ */
+const generateResponseSuccessCheck = (pmOperation: any, oaOperation: any): any => {
+  // console.log('pmOperation', pmOperation)
+  console.log('oaOperation', oaOperation)
+
+  // Add status success check
+  const pmTest: string[] = []
+  pmTest.push('// Validate status 2xx \n')
```

Comment:
wonder if it doesn't make sense to use a template with interpolation for tests. This way we can have a sep dir dedicated for tests that just take their variables to pass in and render out the string